stdlib/src/builtin/int.mojo

```diff
@@ -1099,6 +1148,51 @@ struct Int(
     # Methods
     # ===-------------------------------------------------------------------===#
 
+    fn mod_inverse(self, m: Self) -> Self:
+        """Compute the modular multiplicative inverse of `self` modulo `m`.
+
+        This method uses the extended Euclidean algorithm to find the modular
+        multiplicative inverse.
+        The result `x` satisfies (`self` * `x`) % `m` == 1.
+
+        Args:
+            m: The modulus. Must be greater than 1.
+
+        Returns:
+            The modular multiplicative inverse of `self` modulo `m`, if it exists.
+            Returns 0 if no inverse exists or if an error occurs.
+
+        Note:
+            - If `self` and `m` are not coprime, no modular inverse exists,
+              and the method returns 0.
+        """
+        if m <= 1:
+            # Modulus must be greater than 1.
+            # This should raise an exception.
+            return 0
+
+        var a = self
+        var b = m
+        var x = 0
+        var y = 1
+        var u = 1
+        var v = 0
+
+        while a != 0:
+            var q = b // a
+            x, u = u, x - q * u
+            y, v = v, y - q * v
+            b, a = a, b - q * a
```

Comment:
style: Here and elsewhere: could you please choose more expressive variable names?